agent/agent.cpp

```diff
@@ -203,19 +204,34 @@ Agent::Agent(
 }
 
 
-Device *Agent::findDeviceByUUIDorName(const std::string &aId)
+const Device * Agent::getDeviceByName(const std::string& name) const
 {
-	Device *device = nullptr;
+	auto devPos = m_deviceMap.find(name);
+	if(devPos != m_deviceMap.end())
+		return devPos->second;
+
+	return nullptr;
+}
+
+
+Device * Agent::getDeviceByName(const std::string& name)
+{
+	auto devPos = m_deviceMap.find(name);
+	if(devPos != m_deviceMap.end())
+		return devPos->second;
+
+	return nullptr;
+}
 
-	std::vector<Device *>::iterator it;
 
-	for (it = m_devices.begin(); !device && it != m_devices.end(); it++)
```

Comment:
This iteration was being used for searching, so I replaced it with std::find_if and used a Lambda expression. I think that this makes its intention clearer.